scenario_player/node_support.py

```diff
@@ -389,9 +397,108 @@ def _validate_options(self, options: Dict[str, Any]):
             if option_name not in KNOWN_OPTIONS:
                 raise ScenarioError(f'Unknown option "{option_name}" supplied.')
 
+    @property
+    def nursery(self):
+        assert self._nursery is not None, "Nursery needs to be set before nodes can be started."
+        return self._nursery
+
+    @nursery.setter
+    def nursery(self, value: Nursery):
+        self._nursery = value
+
+
+class SnapshotManager:
+    def __init__(self, scenario_runner: "ScenarioRunner", node_runners: List[NodeRunner]) -> None:
+        self._scenario_runner = scenario_runner
+        self._node_runners = node_runners
+
+    def check_scenario_config(self):
+        if not self._scenario_runner.definition.nodes.reuse_accounts:
+            raise ScenarioError("Snapshots aren't supported when 'nodes.reuse_accounts' is False.")
+
+        token_config = self._scenario_runner.definition.token
+        if (not token_config.should_reuse_token) and token_config.address is None:
+            raise ScenarioError(
+                "Snapshots are only supported when token reuse is enabled "
+                "or a fixed token address is configured."
+            )
+
+    def _check_conditions(self):
+        all_nodes_stopped = all(not node_runner.is_running for node_runner in self._node_runners)
+        assert all_nodes_stopped, "Can't perform snapshot operations while nodes are running."
+        self.check_scenario_config()
+
+    def _get_snapshot_dirs(self, ensure: bool = False) -> List[Path]:
```

Comment:
Is this used with `ensure = True` anywhere?

Reply:
Nope, good catch.